### PR TITLE
Make Code Analysis workflow uniform

### PR DESCRIPTION
**Pull request by szepeviktor**
- upgrade actions/checkout
- quote YAML string
- `composer exec` is for running bins

### CODE REVIEW

File: .github/workflows/code_analysis.yml [hunk 1]
```diff
@@ -15,38 +15,36 @@ jobs:
...
+                        run: 'composer exec -- phpunit'
                     -
                         name: 'Check Commented Code'
-                        run: vendor/bin/easy-ci check-commented-code src tests --ansi
-
+                        run: 'composer exec -- easy-ci check-commented-code src tests --ansi'
                     -
                         name: 'Check Active Classes'
-                        run: vendor/bin/easy-ci check-active-class src --ansi
+                        run: 'composer exec -- easy-ci check-active-class src --ansi'
```
Comment (TomasVotruba) [inline .github/workflows/code_analysis.yml:30]: This is hacking 2 items together, very hard to read.

File: .github/workflows/code_analysis.yml [hunk 2]
```diff
@@ -15,38 +15,36 @@ jobs:
...
                     -
                         name: 'ECS'
-                        run: 'vendor/bin/ecs'
-
+                        run: 'composer run check-cs'
                     -
                         name: 'PHPStan'
-                        run: 'vendor/bin/phpstan'
-
+                        run: 'composer run phpstan'
```
Comment (TomasVotruba) [inline .github/workflows/code_analysis.yml:21]: Composer adds it's own output wrapper and process wrapper, that add one more level of complexity to debug.
The tool setup is in the configs, so no need to wrap them with composer script. 

The goal of CI is to give as fast and as narrow feedback as possible, so we keep the tool calls here.

**Comment by TomasVotruba** [comment]
Thanks for the PR!

I think changing these details is something not a priority at the moment. It adds maintenance and review cost for not much gain.

What would be really useful is to improve the real project testing and test fixture for your failures :+1: That's where the biggest value of this tool is. Could you chech that?

**Comment by szepeviktor** [comment]
You know, Tomas. I understand this _value_ thing but I think an iceberg is build from the bottom where no one sees any value.
Nowadays IT (and other industries) are rapidly deteriorating.

**Comment by szepeviktor** [comment]
I'm thinking of adding e2e test with an existing/popular Laravel app.

**Comment by szepeviktor** [comment]
Maybe https://github.com/monicahq/monica/tree/main/resources/views ...

**Pull request closed without merging**